applications/RANSApplication/python_scripts/formulations/rans_formulation.py

```diff
@@ -1,119 +1,107 @@
 import KratosMultiphysics as Kratos
 import KratosMultiphysics.RANSApplication as KratosRANS
 
-class RansFormulation:
-    def __init__(self, base_computing_model_part, settings):
-        """RansFormulation base class
-
-        This class is the base class for formulations used in RANSApplication. A single leaf formulation
-        is responsible for solving for one variable only. RansFormulations can be added to another RansFormulation
-        creating coupled hierarchical formulation. Then this base RansFormulations solves them recursively according
-        to the addition order.
+## @addtogroup RANSApplication
+## @{
+## @name Kratos classes
+## @{
 
-        Args:
-            base_computing_model_part (Kratos.ModelPart): Base model part, which is copied to create solvers for given formulation
-            settings (Kratos.Parameters): Settings to be used in this formulation
+class RansFormulation:
+    """ @brief RansFormulation base class.
+        @classname RansFormulation
+        @details This class is the base class for formulations used in RANSApplication. A single leaf formulation
+                 is responsible for solving for one variable only. RansFormulations can be added to another RansFormulation
+                 creating coupled hierarchical formulation. Then this base RansFormulations solves them recursively according
+                 to the addition order.
+    """
+
+    def __init__(self, base_computing_model_part: Kratos.ModelPart, settings: Kratos.Parameters):
+        """ @param base_computing_model_part: Base model part, which is copied to create solvers for given formulation.
+            @param settings: Settings to be used in this formulation.
         """
         self.__settings = settings
         self.__base_computing_model_part = base_computing_model_part
         self.__list_of_formulations = []
         self.__list_of_processes = []
         self.__move_mesh = False
 
-    def GetParameters(self):
-        """Returns parameters used in this formulation
-
-        Returns:
-            Kratos.Parameters: Parameters of this formulation
+    def GetParameters(self) -> Kratos.Parameters:
+        """ @brief Returns parameters used in this formulation.
+            @return Parameters of this formulation.
         """
         return self.__settings
 
-    def GetDomainSize(self):
-        """Returns domain size
-        """
+    def GetDomainSize(self) -> int:
+        """ @brief Returns domain size."""
         return self.__base_computing_model_part.ProcessInfo[Kratos.DOMAIN_SIZE]
 
-    def AddRansFormulation(self, formulation):
-        """Adds another RansFormulation to current formulation creating a list of formulations.
-
-        Args:
-            formulation (RansFormulation): Formulation to be added
+    def AddRansFormulation(self, formulation: "RansFormulation") -> None:
```

Comment:
```suggestion
    def AddRansFormulation(self, formulation: RansFormulation) -> None:
```

I recommend to use it directly

Reply:
the python interpreter reads member signatures when the class isn't fully defined yet, so I can't use the class in there directly, only in the function body.